### PR TITLE
Unroll to the given basis

### LINKED ISSUE #2057: Transpile unrolls to U gates even if gate is specified in basis_gates
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```
q1 = QuantumRegister(1,'q1')
q2 = QuantumRegister(2,'q2')
q3 = QuantumRegister(3,'q3')
qc = QuantumCircuit(q1, q2, q3)
qc.h(q1[0])
qc.h(q2[1])
qc.h(q3[2])
layout = [4, 5, 6, 8, 9, 10]

cmap = [[1, 0], [1, 2], [2, 3], [4, 3], [4, 10], [5, 4], [5, 6], [5, 9],
 [6, 8], [7, 8], [9, 8], [9, 10], [11, 3], [11, 10], [11, 12], [12, 2], [13, 1],
 [13, 12]]

new_circ = transpile(qc, backend=None,
                     coupling_map=cmap,
                     basis_gates=['h'],
                     initial_layout=layout)
new_circ.data
```
```
[<qiskit.extensions.standard.u2.U2Gate at 0xd1c5cbcf8>,
 <qiskit.extensions.standard.u2.U2Gate at 0xd1c5cbe48>,
 <qiskit.extensions.standard.u2.U2Gate at 0xd1c5cbf60>]
```

### What is the expected behavior?
I would expect that defining a gate to be in the basis gates would not unroll to the U gate equivalent.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #2057 

This PR  unrolls to the given basis. Because our default optimization pass requires the "standard" basis, it wont run in that case.

### CODE REVIEW

File: qiskit/transpiler/preset_passmanagers/default.py [hunk 1]
```diff
@@ -71,15 +71,17 @@ def default_pass_manager(basis_gates, coupling_map, initial_layout, seed_transpi
...
-    pass_manager.append(Unroller(['u1', 'u2', 'u3', 'id', 'cx']))
-
-    # Simplify single qubit gates and CXs
-    simplification_passes = [Optimize1qGates(), CXCancellation(), RemoveResetInZeroState()]
-
+    # simplifications and optimizations
+    simplification_passes = [RemoveResetInZeroState(), CXCancellation()]
+    if set(basis_gates).issubset(set(['u1', 'u2', 'u3', 'id', 'cx', 'U', 'CX'])):
+        # Simplify single qubit gates
+        simplification_passes += [Optimize1qGates()]
```
Comment (jaygambetta) [inline qiskit/transpiler/preset_passmanagers/default.py]: why U C

Comment (jaygambetta) [reply qiskit/transpiler/preset_passmanagers/default.py]: remove?

**Comment by ajavadia** [comment]
I would do this differently.  Instead of conditionally invoking the `Optimize1qGates`, I think we should just remove this line from that pass:
```
        self.requires.append(Unroller(["u1", "u2", "u3", "cx", "id"]))
```
I think this was a bad choice in the beginning. It basically causes the pass to do something beyond what it is designed to do. It is supposed to merge single qubit rotations whenever it can, not unroll the circuit.

File: qiskit/transpiler/preset_passmanagers/default.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/transpiler/preset_passmanagers/default.py]: because `Optimize1qGates()` also works on `U` and `CX`.

**Comment by 1ucian0** [comment]
The pass would not do anything in that case.

File: qiskit/transpiler/preset_passmanagers/default.py (hunk 1 above)
Comment (jaygambetta) [reply qiskit/transpiler/preset_passmanagers/default.py]: it really should not. These are ground truths for the matrix. We really should not be using them anywhere in Qiskit.

Comment (1ucian0) [reply qiskit/transpiler/preset_passmanagers/default.py]: Ups. I was wrong. `Optimize1qGates` does not work on `U` or `CX`. Fixed in 653995d0 
```
In:             qr = QuantumRegister(1, 'qr') 
   ...:         circuit = QuantumCircuit(qr) 
   ...:         circuit.u_base(1,2,3,qr) 
   ...:         circuit.u_base(2,2,1,qr)  
   ...:         dag = circuit_to_dag(circuit) 
   ...:         expected = QuantumCircuit(qr) 
   ...:  
   ...:         pass_ = Optimize1qGates() 
   ...:         after = pass_.run(dag) 
   ...:  
   ...:         print(dag_to_circuit(after))                                                                                                           
         ┌──────────┐┌──────────┐
qr_0: |0>┤ U(1,2,3) ├┤ U(2,2,1) ├
         └──────────┘└──────────┘
```

**Comment by ajavadia** [comment]
I did this in #2266, added the same tests as here. The difference is that now there is no hardcoded Unrolling to `u1, u2, u3, cx` anywhere. It just unrolls to the basis everywhere. So the job of `Optimize1qGates` is isolated.

**Pull request closed without merging**